working/control-flow-collections/implementation-plan.md

```diff
@@ -0,0 +1,189 @@
+# Implementation Plan for "Control Flow Collections"
+
+Owner: rnystrom@google.com ([@munificent](https://github.com/munificent/) on GitHub)
+
+Relevant links:
+
+* [Tracking issue](https://github.com/dart-lang/language/issues/78)
+* [Proposal](https://github.com/dart-lang/language/blob/master/working/control-flow-collections/feature-specification.md)
+
+## Phase 0 (Prerequisite)
+
+### "control-flow-collections" Experimental flag
+
+The implementation of this feature should be hidden behind an [experiment
+flag][]. Tools must be passed the flag
+`--enable-experiment=control-flow-collections` to enable the feature.
+
+[experiment flag]: https://github.com/dart-lang/sdk/blob/master/docs/process/experimental-flags.md
+
+While this feature is under development, individual tools may have incomplete or
+changing implementations behind the flag. When all tools have completely
+implemented the feature, the the feature will be enabled by default, and the
+flag removed in a stable release.
+
+### Tests
+
+The language team adds tests for the feature.
+
+## Phase 1 (Foundation)
+
+### CFE
+
+The CFE implements parsing the new syntax, type checking it, and compiling it to
+Kernel. Since the CFE will be implementing constant evaluation, it also
+implements evaluating `if` and `for` in constant collections.
+
+This feature can likely be implemented entirely in the front end, so back-end
+support may not be needed. If it does require Kernel changes, the back end will
+need to handle those changes.
+
+### Analyzer
+
+The analyzer implements parsing the new syntax, type checking it, and
+evaluating `if` and `for` in constant collections.
+
+## Phase 2 (Tool Implementation)
+
+### dart2js
+
+If the feature is handled by the front end, there may be no dart2js work.
+Otherwise, dart2js may need to handle any Kernel changes or otherwise add
+support for this.
+
+### Dartfmt
+
+Define and implement formatting rules for the new syntax. Add formatting tests.
+
+### DDC
+
+If this feature can be implemented entirely in the front end with no Kernel
+changes, and DDC is entirely onto the CFE, then no DDC changes may be needed.
+Otherwise, DDC may need to handle any Kernel changes or otherwise add support
+for this.
+
+DDC may need to support canonicalizing constant collections with spread
+operators.
+
+### IntelliJ/Grok
+
+Update to use the latest analyzer with support for the feature. Likely no other
+changes explicitly needed. There a are a handful of usability features that
```

Comment:
Intellij has its own parser which needs updating.  Also github markdown.

Reply:
Done. For the latter, I'm guessing you mean: https://github.com/dart-atom/dart, which is what drives Dart syntax highlighting on GitHub.

Reply:
Also https://github.com/codemirror/CodeMirror/blob/master/mode/dart/dart.js ?

Reply:
Not sure. @devoncarew?

Reply:
The codemirror and atom grammars are largely regex based. My guess is that there won't be any changes required to support the control flow work.